network/p2p/test/topic_validator_test.go

```diff
@@ -260,14 +284,31 @@ func TestTopicValidator_InvalidTopic(t *testing.T) {
 
 	timedCtx, cancel5s := context.WithTimeout(ctx, 5*time.Second)
 	defer cancel5s()
-	// create a dummy block proposal to publish from our SN node
-	data1 := p2pfixtures.MustEncodeEvent(t, unittest.ProposalFixture(), channels.PushBlocks)
 
-	err = sn2.Publish(timedCtx, topic, data1)
+	// invalid topic is malformed, hence it cannot be used to create a message scope, as it faces an error.
+	// Hence, we create a dummy block proposal message scope to publish on a legit topic, and then override
+	// the topic in the next step to a malformed topic.
+	dummyMessageScope, err := message.NewOutgoingScope(
+		flow.IdentifierList{identity1.NodeID, identity2.NodeID},
+		channels.TopicFromChannel(channels.PushBlocks, sporkId),
+		unittest.ProposalFixture(),
+		unittest.NetworkCodec().Encode,
+		message.ProtocolTypePubSub)
+	require.NoError(t, err)
+
+	// overrides the topic to be an invalid topic
+	corruptOutgoingMessageScope := mocknetwork.NewOutgoingMessageScope(t)
+	corruptOutgoingMessageScope.On("TargetIds").Return(dummyMessageScope.TargetIds()).Maybe()
+	corruptOutgoingMessageScope.On("Topic").Return(topic).Maybe()
+	corruptOutgoingMessageScope.On("Proto").Return(dummyMessageScope.Proto()).Maybe()
+	corruptOutgoingMessageScope.On("PayloadType").Return(dummyMessageScope.PayloadType()).Maybe()
+	corruptOutgoingMessageScope.On("Size").Return(dummyMessageScope.Size()).Maybe()
```

Comment:
```suggestion
	corruptOutgoingMessageScope.On("Topic").Return(topic)
	corruptOutgoingMessageScope.On("Proto").Return(dummyMessageScope.Proto())
	corruptOutgoingMessageScope.On("PayloadType").Return(dummyMessageScope.PayloadType())
	corruptOutgoingMessageScope.On("Size").Return(dummyMessageScope.Size())
```

Reply:
I ran the test without all the `Maybe()` calls with `-count 10` and it passed, so I think we should remove the `Maybe()` calls which make the test less meaningful if all those calls aren't required.

Reply:
https://github.com/onflow/flow-go/pull/4635/commits/701e14d8dd940b6da2749925d6c01b858bf3cdd1